src/App.tsx

```diff
@@ -168,12 +185,19 @@ function App() {
   }, [nextPageState]);
 
   const getMachineState = useCallback(async () => {
-    const resp = await fetch(`${ATM_BACKEND_URL}/stats`).then((x) => {
-      return x.json()
+    if (!recipientAddress) {
+      const resp = await fetch(`${ATM_BACKEND_URL}/stats`).then((x) => x.json());
+      setMachineState(resp as MachineState);
+      return;
     }
-    );
+
+    const resp = await fetch(`${ATM_BACKEND_URL}/stats?address=${recipientAddress}`).then((x) => x.json());
```

Comment:
nit: no need for address here, we just use it once and pass in data with setNextState